ontology/rml-core.owl

```diff
@@ -0,0 +1,517 @@
+              rdfs:comment "Represents a predicate map."@en ;
+              rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+              rdfs:label "Predicate Map" .
+
+
+###  http://w3id.org/rml/PredicateObjectMap
+:PredicateObjectMap rdf:type owl:Class ;
+                    rdfs:subClassOf owl:Thing ,
+                                    [ rdf:type owl:Restriction ;
+                                      owl:onProperty :predicateMap ;
+                                      owl:minQualifiedCardinality "1"^^xsd:nonNegativeInteger ;
+                                      owl:onClass :PredicateMap
+                                    ] ,
+                                    [ rdf:type owl:Restriction ;
+                                      owl:onProperty :objectMap ;
+                                      owl:qualifiedCardinality "1"^^xsd:nonNegativeInteger ;
+                                      owl:onClass :ObjectMap
+                                    ] ;
+                    rdfs:comment "Represents a predicate-object map."@en ;
+                    rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+                    rdfs:label "Predicate Object Map" .
+
+
+###  http://w3id.org/rml/RefObjectMap
+:RefObjectMap rdf:type owl:Class ;
+              rdfs:subClassOf owl:Thing ;
+              rdfs:comment "Denotes a reference to an object map."@en ;
+              rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+              rdfs:label "Reference Object Map" .
+
+
+###  http://w3id.org/rml/StarMap
+:StarMap rdf:type owl:Class ;
+         rdfs:comment "Represents a star map" ;
+         rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+         rdfs:label "Star Map" .
+
+
+###  http://w3id.org/rml/SubjectMap
+:SubjectMap rdf:type owl:Class ;
+            rdfs:subClassOf :TermMap ,
+                            [ rdf:type owl:Restriction ;
+                              owl:onProperty :class ;
+                              owl:minQualifiedCardinality "0"^^xsd:nonNegativeInteger ;
+                              owl:onClass rdfs:Class
+                            ] ;
+            rdfs:comment "Represents a subject map."@en ;
+            rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+            rdfs:label "Subject Map" .
+
+
+###  http://w3id.org/rml/TermMap
+:TermMap rdf:type owl:Class ;
+         rdfs:subClassOf :ExpressionMap ;
+         rdfs:comment "A function that generates an RDF term from a logical table row."@en ;
+         rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+         rdfs:label "Term Map"@en .
+
+
+###  http://w3id.org/rml/TriplesMap
+:TriplesMap rdf:type owl:Class ;
+            rdfs:subClassOf owl:Thing ,
+                            [ rdf:type owl:Restriction ;
+                              owl:onProperty :subjectMap ;
+                              owl:qualifiedCardinality "1"^^xsd:nonNegativeInteger ;
+                              owl:onClass :SubjectMap
+                            ] ;
```

Comment:
We can also have a `:TriplesMap` without a declared subject map, right?

Reply:
removed -- along with the other cardinality restrictions

Reply:
> We can also have a :TriplesMap without a declared subject map, right?

Can we ?